spec/jobs/state_file/client_interaction_notification_email_job_spec.rb

```diff
@@ -0,0 +1,124 @@
+      allow(mailer).to receive(:deliver_now).and_return(Mail::Message.new(message_id: message_id))
+      allow(InternalEmail).to receive(:create!).and_call_original
+    end
+
+    context "when hub_email_notifications flipper flag is enabled and last client message is unanswered" do
+      before do
+        allow(Flipper).to receive(:enabled?).with(:hub_email_notifications).and_return(true)
+      end
+
+      context "one interaction within 10 minutes" do
+        it "sends the message using deliver_now and persists the message_id & sent_at" do
+          Timecop.freeze(fake_time) { described_class.perform_now(interaction, user, received_at: fake_time) }
+
+          internal = InternalEmail.last
+          expect(UserMailer).to have_received(:incoming_interaction_notification_email)
+                                  .with(internal.deserialized_mail_args)
+          expect(internal.reload.outgoing_message_status.message_id).to eq(message_id)
+        end
+
+        it "deletes the client interaction" do
+          expect do
+            described_class.perform_now(interaction, user)
+          end.to change(ClientInteraction, :count).by -1
+        end
+      end
+
+      context "when not the most recent interaction within 10 minutes" do
+        let!(:interaction_2) { create(:client_interaction, client: client, interaction_type: "new_client_message", created_at: fake_time + 3.minutes) }
+        let!(:interaction_3) { create(:client_interaction, client: client, interaction_type: "new_client_message", created_at: fake_time + 5.minutes) }
+
+        it "doesn't send the message and doesn't delete any ClientInteractions" do
+          expect do
+            described_class.perform_now(interaction, user)
+          end.to change(ClientInteraction, :count).by 0
+
+          expect(UserMailer).not_to have_received(:incoming_interaction_notification_email)
+        end
+      end
+
+      context "2 interactions within 10 minutes and 1 interaction after the interaction window" do
+        let!(:interaction_2) { create(:client_interaction, client: client, interaction_type: "new_client_message", created_at: fake_time + 3.minutes) }
+        let!(:interaction_3) { create(:client_interaction, client: client, interaction_type: "new_client_message", created_at: fake_time + 11.minutes) }
+
+        it "sends the messsage for first window and deletes their interactions and not the second window" do
+          expect do
+            described_class.perform_now(interaction_2, user)
+          end.to change(ClientInteraction, :count).by -2
+          expect(UserMailer).to have_received(:incoming_interaction_notification_email)
+
+          expect(ClientInteraction.find_by(id: interaction.id)).to be_nil
+          expect(ClientInteraction.find_by(id: interaction_2.id)).to be_nil
+          expect(ClientInteraction.find_by(id: interaction_3.id)).to be_present
+        end
```

Comment:
I think I might have misunderstood the requirement -- I thought we'd wait until the third interaction triggers the job & scoop up all remaining interactions